app/controllers/registrations_controller.rb

```diff
@@ -3,7 +3,8 @@ class RegistrationsController < Devise::RegistrationsController
   private
 
   def sign_up_params
-    params.require(:user).permit(:name, :email, :password, :password_confirmation)
+    password = SecureRandom.hex(10)
```

Comment:
One lesson that one can extract is that workaround code is rarely necessary. Attack causes not symptoms!